Configure OBS instance to perform Clang builds

athos-ribeiro opened the pull request:
This commit ports the salt states available at
https://github.com/athos-ribeiro/salt-obs to provision
irill8.siege.inria.fr with capabilities to monitor Debian
archives and trigger builds for newly accepted packages
substituting GCC for Clang binaries.

athos-ribeiro committed:
Configure OBS instance to perform Clang builds

This commit ports the salt states available at
https://github.com/athos-ribeiro/salt-obs to provision
irill8.siege.inria.fr with capabilities to monitor Debian
archives and trigger builds for newly accepted packages
substituting GCC for Clang binaries.

sylvestre commented on README.md:
```diff
@@ -0,0 +1,17 @@
+# Open Build Service at irill8.siege.inria.fr
```
This salt instance is doing other things. Please add something on the top of this file!

athos-ribeiro replied:
Done

sylvestre commented on obs-server.sls:
```diff
@@ -1,4 +1,21 @@
-install obs server packages:
+/etc/apt/sources.list:
```
I don't think we want to do that. We should instead add the repo into 
/etc/apt/sources.list.d/ instead

sylvestre replied:
and you should use https://docs.saltstack.com/en/latest/ref/states/all/salt.states.pkgrepo.html

athos-ribeiro replied:
OK. I am looking into it

sylvestre replied:
ping?

sylvestre commented on obs-server.sls:
```diff
@@ -1,4 +1,21 @@
+  host.present:
+    - ip: 127.0.0.1
+
+refresh_packages_db:
```
This isn't necessary, add a repo the salt way will do it
https://docs.saltstack.com/en/latest/ref/states/all/salt.states.pkgrepo.html

sylvestre replied:
ping?

sylvestre commented on obs-server.sls:
```diff
@@ -22,11 +39,141 @@ install apache:
+      - obs-build
+    - fromrepo: stretch-backports
+
+install_libsolv_from_testing:
```
Please add a comment (or link to your blog post) explaining why we are doing that.

athos-ribeiro replied:
Done

sylvestre commented on obs-server.sls:
```diff
@@ -22,11 +39,141 @@ install apache:
+      - libsolvext0
+    - fromrepo: buster
+
+/usr/share/obs/api/Gemfile:
```
ditto

athos-ribeiro replied:
Done

sylvestre commented on obs-server/check_new_uploads:
```diff
@@ -0,0 +1,75 @@
+    subprocess.run([TRIGGER_BIN_PATH, pkg['name']])
+
+
+subprocess.run(['apt', 'update'])
```
why do you have to do it?

athos-ribeiro replied:
As of now, we are checking emails every 2 hours. Don't we need to update the cache to fetch packages that were just uploaded?

sylvestre replied:
This requires to be root. I would prefer you use sudo for that

sylvestre commented on obs-server/check_new_uploads:
```diff
@@ -0,0 +1,75 @@
+        if (pkg['dist'] in ['unstable', 'sid']) and 'source' in pkg['arch']:
+            trigger_build(pkg)
+    # Uncomment the following lines if you want to remove the email afterwards
+    # M.store(num, '+FLAGS', '\\Deleted')
```
why we are not doing that? (removing the email)

athos-ribeiro replied:
No specific reason. I Just uncommented the lines :)

sylvestre committed:
Update of the description

sylvestre commented on obs-server/obs_instance_configuration.xml:
```diff
@@ -0,0 +1,31 @@
+<configuration>
+  <title>Debian LLVM OBS builder</title>
+  <description>  &lt;p class="description"&gt;
+		The Debian LLVM OBS builder is used to both distribute upstream LLVM .deb packages
```
you can trim the description just to keep your part!

athos-ribeiro replied:
Done

sylvestre commented on obs-server/obs_instance_configuration.xml:
```diff
@@ -0,0 +1,31 @@
+  <disable_publish_for_branches>on</disable_publish_for_branches>
+  <admin_email>unconfigured@openbuildservice.org</admin_email>
+  <schedulers>
+    <arch>armv7l</arch>
```
you are also remove arm devices

athos-ribeiro replied:
Done

sylvestre commented on obs-server/sources.list:
```diff
@@ -0,0 +1,7 @@
+deb http://httpredir.debian.org/debian stretch main
```
please remove this file

sylvestre replied:
ping?

sylvestre commented on obs-server/trigger_clang_build:
```diff
@@ -0,0 +1,55 @@
+    echo 'Preparing obs-service-clang-build package...'
+    mkdir obs-service-clang-build
+    cd obs-service-clang-build/
+    wget 'https://www.ime.usp.br/~athoscr/files/obs/obs-service-clang-build_0.1-1.debian.tar.xz'
```
please add this to this repo and use that. I would like to avoid external dependencies

athos-ribeiro replied:
is it ok to add the tarballs here then?

sylvestre replied:
yeah, better than on thirdparty website.

athos-ribeiro replied:
Done.

athos-ribeiro committed:
Update README file to describe repo features

athos-ribeiro committed:
Remove non-implemented feature from description

athos-ribeiro committed:
Document reasons for files/libs substitutions

athos-ribeiro committed:
Remove emails after triggering Clang builds

athos-ribeiro committed:
Remove arm schedulers

athos-ribeiro committed:
Include obs-service-clang-build package in salt fs

We rather not depend on external links to fetch our dependencies, hence
we add the package sources in our repository.

athos-ribeiro committed:
Use proper salt states to configure repositories

athos-ribeiro committed:
Set higher preference for Debian Stable

Since we are enabling backports and testing repositories, we want
stretch to have a higher preference so other packages are installed in
their stable version.

athos-ribeiro committed:
Work with single packages upon triggering builds
